src/arch/mod.rs

```diff
@@ -1,3 +1,8 @@
+#![allow(
+	clippy::allow_attributes,
+	reason = "Required for arch-specific code (dead code)"
+)]
```

Comment:
What is the reason for this allow? Did it trigger a warning?

Reply:
I count three allows in `src/arch` :wink:

Reply:
I understand that the complexity of dealing with this change has increased / become potentially overwhelming for us all, but I think this might very well actually illustrate the necessity here.